default/AI/AIDependencies.py

```diff
@@ -88,6 +95,71 @@
+                "CON_ART_HEAVENLY", "CON_ART_PLANET",
+                "SHP_NOVA_BOMB", "SHP_DEATH_SPORE", "SHP_BIOTERM"]
+
+THEORY_TECHS = ["LRN_PHYS_BRAIN", "LRN_TRANSLING_THT", "LRN_PSIONICS", "LRN_GRAVITONICS", "LRN_EVERYTHING", "LRN_MIND_VOID", "LRN_NDIM_SUBSPACE", "LRN_TIME_MECH",
+                "GRO_PLANET_ECOL", "GRO_GENETIC_ENG", "GRO_ADV_ECOMAN", "GRO_NANOTECH_MED", "GRO_TRANSORG_SENT",
+                "PRO_NANOTECH_PROD", "PRO_ZERO_GEN",
+                "CON_ASYMP_MATS", "CON_ARCH_PSYCH",
+                "SHP_GAL_EXPLO"]
+
+DEFENSE_TECHS_PREFIX = "DEF"
+
+PRODUCTION_BOOST_TECHS = ["PRO_ROBOTIC_PROD", "PRO_FUSION_GEN", "PRO_SENTIENT_AUTOMATION",
+                          "PRO_INDUSTRY_CENTER_I", "PRO_INDUSTRY_CENTER_II", "PRO_INDUSTRY_CENTER_III",
+                          "PRO_SOL_ORB_GEN"]
+
+RESEARCH_BOOST_TECHS = ["LRN_ALGO_ELEGANCE", "LRN_ARTIF_MINDS", "LRN_DISTRIB_THOUGHT", "LRN_QUANT_NET", "LRN_STELLAR_TOMOGRAPHY",
+                        "LRN_ENCLAVE_VOID"]
+
+PRODUCTION_AND_RESEARCH_BOOST_TECHS = ["LRN_UNIF_CONC", "GRO_ENERGY_META"]
+
+POPULATION_BOOST_TECHS = ["GRO_SYMBIOTIC_BIO", "GRO_XENO_HYBRIDS", "GRO_CYBORG", "GRO_SUBTER_HAB",
+                          "CON_ORBITAL_HAB", "CON_NDIM_STRC", "PRO_EXOBOTS"]
+
+SUPPLY_BOOST_TECHS = ["CON_ARCH_MONOFILS", "CON_GAL_INFRA", "CON_CONTGRAV_ARCH", "CON_ORBITAL_CON"]
+
+METER_CHANGE_BOOST_TECHS = ["CON_FRC_ENRG_STRC", "CON_TRANS_ARCH"]
+
+DETECTION_TECHS = ["SPY_DETECT_1", "SPY_DETECT_2", "SPY_DETECT_3", "SPY_DETECT_4", "SPY_DETECT_5", "SPY_DIST_MOD", "SPY_LIGHTHOUSE"]
+STEALTH_TECHS = ["SPY_STEALTH_1", "SPY_STEALTH_2", "SPY_STEALTH_3", "SPY_STEALTH_4", "CON_FRC_ENRG_CAMO"]
+
+ROBOTIC_HULL_TECHS = ["SHP_MIL_ROBO_CONT", "SHP_SPACE_FLUX_DRIVE", "SHP_TRANSSPACE_DRIVE", "SHP_CONTGRAV_MAINT", "SHP_MASSPROP_SPEC",
+                      "SHP_NANOROBO_MAINT", "SHP_MIDCOMB_LOG"]
+ASTEROID_HULL_TECHS = ["SHP_ASTEROID_HULLS", "SHP_SCAT_AST_HULL",
+                       "SHP_HEAVY_AST_HULL", "SHP_CAMO_AST_HULL", "SHP_MINIAST_SWARM"]
+ORGANIC_HULL_TECHS = ["SHP_ORG_HULL", "SHP_MULTICELL_CAST", "SHP_ENDOCRINE_SYSTEMS", "SHP_CONT_BIOADAPT",
+                      "SHP_MONOCELL_EXP", "SHP_CONT_SYMB", "SHP_BIOADAPTIVE_SPEC", "SHP_ENDOSYMB_HULL", "SHP_SENT_HULL"]
+ENERGY_HULL_TECHS = ["SHP_FRC_ENRG_COMP", "SHP_QUANT_ENRG_MAG", "SHP_ENRG_BOUND_MAN", "SHP_SOLAR_CONT"]
+MISC_HULL_TECHS = ["SHP_XENTRONIUM_HULL"]
+
+HULL_TECHS = ROBOTIC_HULL_TECHS + ASTEROID_HULL_TECHS + ORGANIC_HULL_TECHS + ENERGY_HULL_TECHS + MISC_HULL_TECHS
+
+DAMAGE_CONTROL_TECHS = ["SHP_BASIC_DAM_CONT", "SHP_FLEET_REPAIR", "SHP_ADV_DAM_CONT"]
+
+WEAPON_TECHS = ["SHP_WEAPON_1_2", "SHP_WEAPON_1_3", "SHP_WEAPON_1_4",
+                "SHP_WEAPON_2_1", "SHP_WEAPON_2_2", "SHP_WEAPON_2_3", "SHP_WEAPON_2_4",
+                "SHP_WEAPON_3_1", "SHP_WEAPON_3_2", "SHP_WEAPON_3_3", "SHP_WEAPON_3_4",
+                "SHP_WEAPON_4_1", "SHP_WEAPON_4_2", "SHP_WEAPON_4_3", "SHP_WEAPON_4_4"]
```

Comment:
It might make sense to use the same prefix approach as you did with defense techs. Not sure about what future plans for names there are though but the naming is quite consistent so far.